bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/Decryption/DecryptWithShares.cs

```diff
@@ -14,29 +15,85 @@ public static class DecryptWithSharesExtensions
     public static PlaintextTally Decrypt(
         this CiphertextTally self,
         List<Tuple<ElectionPublicKey, TallyShare>> guardianShares,
+        List<GuardianChallengeResponse> challengeResponses,
         bool skipValidation = false)
     {
         var lagrangeCoefficients = guardianShares.ComputeLagrangeCoefficients();
         return self.Decrypt(
-            guardianShares, lagrangeCoefficients, skipValidation
+            guardianShares,
+            challengeResponses,
+            lagrangeCoefficients,
+            skipValidation
         );
     }
 
     public static PlaintextTally Decrypt(
         this CiphertextTally self,
         List<Tuple<ElectionPublicKey, TallyShare>> guardianShares,
+        List<GuardianChallengeResponse> challengeResponses,
         Dictionary<string, LagrangeCoefficient> lagrangeCoefficients,
         bool skipValidation = false)
     {
         var accumulation = self.AccumulateShares(
             guardianShares, lagrangeCoefficients, skipValidation);
 
+        accumulation!.AddProofs(
+            self,
+            challengeResponses,
+            skipValidation
+        );
+
+        return self.Decrypt(accumulation, skipValidation);
+    }
+
+    public static PlaintextTally DecryptNoProofs(
```

Comment:
but decrypting without the proofs is still a valid thing to do, particularly as part of unit tests, so those methods that do not require the challenge responses are now named differently